File: swimming-contest-software/Server/src/main/java/server/network/AbstractConcurrentServer.java
package server.network;

import java.net.Socket;

public abstract class AbstractConcurrentServer extends AbstractServer{
    public AbstractConcurrentServer(int port) {
        super(port);
    }

    public void processRequest(Socket client) {
        Thread thread = createWorker(client);
        thread.start();
    }

    protected abstract Thread createWorker(Socket client);
}

File: Server/Network/AbstractConcurrentServer.cs
using System.Net.Sockets;
using System.Threading;

namespace Server.Network;

public abstract class AbstractConcurrentServer: AbstractServer
{
    protected AbstractConcurrentServer(string host, int port) : base(host, port)
    {
    }

    public override void ProcessRequest(TcpClient client)
    {
        Thread t = createWorker(client);
        t.Start();
    }

    protected abstract Thread createWorker(TcpClient client);
}